File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        if len(s) != len(t):
            return False
        
        counter = [0] * 26

        for i in range(len(s)):
            counter[ord(s[i]) - ord('a')] = counter[ord(s[i]) - ord('a')] + 1
            counter[ord(t[i]) - ord('a')] = counter[ord(t[i]) - ord('a')] - 1

        for count in counter:
            if count != 0:
                return False
    
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) 
        return false;

        int[] scounter = new int[26];
        int[] tcounter = new int[26];

        for(int i=0; i<s.length(); i++){
           scounter[s.charAt(i) - 'a']++;
           tcounter[t.charAt(i) - 'a']++;     
        }

        for (int i = 0; i < 26; i++) {
            if(scounter[i] != tcounter[i])
                return false;
        }

        return true;

        
    }
}